Dockerfile

```diff
@@ -0,0 +1,106 @@
+# Build:
+#   docker build -t discourse/ruby:3.3-bookworm-slim --build-arg DEBIAN_RELEASE=bookworm --build-arg RUBY_VERSION=3.3.3 .
+#
+# You are not expected to build and push this Docker image by hand. This is done by the `build` Github Actions workflow.
+ARG DEBIAN_RELEASE=
+ARG RUBY_VERSION=
+
+FROM debian:${DEBIAN_RELEASE}-slim
+
+ARG RUBY_VERSION
+
+ENV LANG en_US.UTF-8 \
+  LANGUAGE en_US.UTF-8 \
+  LC_ALL en_US.UTF-8 \
+  RUBY_VERSION=3.3.3
+
+# Installs system dependencies required to run Ruby
+RUN set -eux; \
+  DEBIAN_FRONTEND=noninteractive apt-get update; \
+  DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  bzip2 \
+  ca-certificates \
+  libgmp-dev \
+  libssl-dev \
+  libyaml-dev \
+  procps \
+  zlib1g-dev \
+  ; \
+  rm -rf /var/lib/apt/lists/*; \
+  # skip installing gem documentation
+  mkdir -p /usr/local/etc; \
+  { \
+  echo 'install: --no-document'; \
+  echo 'update: --no-document'; \
+  } >> /usr/local/etc/gemrc; \
+  # Installs the dependencies required to build Ruby
+  savedAptMark="$(apt-mark showmanual)"; \
+  DEBIAN_FRONTEND=noninteractive apt-get update; \
+  DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  git \
+  dpkg-dev \
+  libgdbm-dev \
+  ruby \
+  autoconf \
+  g++ \
+  gcc \
+  libbz2-dev \
+  libgdbm-compat-dev \
+  libglib2.0-dev \
+  libncurses-dev \
+  libxml2-dev \
+  libxslt-dev \
+  make \
+  wget \
+  xz-utils \
+  ; \
+  rm -rf /var/lib/apt/lists/*; \
+  # Install Rust to build Ruby with YJIT
+  rustArch=; \
+  dpkgArch="$(dpkg --print-architecture)"; \
+  case "$dpkgArch" in \
+  'amd64') rustArch='x86_64-unknown-linux-gnu'; rustupUrl='https://static.rust-lang.org/rustup/archive/1.26.0/x86_64-unknown-linux-gnu/rustup-init'; rustupSha256='0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db' ;; \
+  'arm64') rustArch='aarch64-unknown-linux-gnu'; rustupUrl='https://static.rust-lang.org/rustup/archive/1.26.0/aarch64-unknown-linux-gnu/rustup-init'; rustupSha256='673e336c81c65e6b16dcdede33f4cc9ed0f08bde1dbe7a935f113605292dc800' ;; \
+  *) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
+  esac; \
+  mkdir -p /tmp/rust; \
+  wget -O /tmp/rust/rustup-init "$rustupUrl"; \
+  echo "$rustupSha256 */tmp/rust/rustup-init" | sha256sum --check --strict; \
+  chmod +x /tmp/rust/rustup-init; \
+  export RUSTUP_HOME='/tmp/rust/rustup' CARGO_HOME='/tmp/rust/cargo'; \
+  export PATH="$CARGO_HOME/bin:$PATH"; \
+  /tmp/rust/rustup-init -y --no-modify-path --profile minimal --default-toolchain '1.77.0' --default-host "$rustArch"; \
+  rustc --version; \
+  cargo --version; \
```

Comment:
Stole this from https://github.com/docker-library/ruby/blob/b8b0053a92b4f7232fe28c45e44a469c032affe2/3.3/slim-bookworm/Dockerfile#L63-L83